src/bmp.cpp

```diff
@@ -558,23 +553,87 @@ int BMP::calculate_colour_count(const BMP& base, Colour to_compare)
     return colour_count;
 }
 
-const std::vector<bool>& BMP::get_blurred_edge_mask() const {
-    return m_blurred_edge_mask;
-}
-const std::vector<bool> BMP::get_vertical_edge_mask() const {
-    return get_vertical_edges<245>();
+
+std::vector<uint8_t> BMP::calculate_intersection_mask(const BMP &original, const BMP &target)
+{
+    const std::vector<uint8_t> original_edge_map = original.get_blurred_edge_mask();
+    const std::vector<uint8_t> target_edge_map = target.get_blurred_edge_mask();
+
+    int width = std::min(original.get_width(), target.get_width());
+    int height = std::min(original.get_height(), target.get_height());
+
+    std::vector<uint8_t> intersection_mask(width * height, 0);
+    for (int i = 0; i < width * height; i++)
+    {
+        intersection_mask[i] = original_edge_map[i] && target_edge_map[i];
```

Comment:
I had accidentally commented about this at <https://github.com/CollaboraOnline/mso-test/commit/3a47f5b2ee833f3a80ceed0f012e32390e6f138b#r164625979>, where you presumably didn't notice it (Github PR review really is a pain):  "it might even be cheaper then to replace short-circuiting `operator&&` with bitwise `operator&`"